### PR TITLE
WP-366 fix: minor changes to the design and functionality

### PR DESCRIPTION
**Pull request by yasgreen93**
Gary commented in the ticket here https://jira.economist.com/browse/WP-366 about some design changes that were missing, this fixes them.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -222,6 +224,16 @@ export default class BlogPost extends React.Component {
     }
   }
 
+  generateBlogPostFlyTitle(siblingListTitle, flyTitle) {
+    let blogFlyTitle = flyTitle;
+    if (this.props.showSiblingArticlesList && siblingListTitle === flyTitle) {
```
Comment (varholak-peter) [inline src/index.js]: This is unnecessary as `blogFlyTitle` is already set.

File: src/index.js [hunk 2]
```diff
@@ -327,7 +340,7 @@ export default class BlogPost extends React.Component {
         {articleHeader}
         <TitleComponent
           title={this.props.title}
-          flyTitle={this.props.flyTitle}
+          flyTitle={this.generateBlogPostFlyTitle(this.props.siblingListSideTitle, this.props.flyTitle)}
```
Comment (varholak-peter) [inline src/index.js]: You can destructure `this.props.siblingListSideTitle` and `flytitle` is already destructured. Makes it a bit clearer.

File: src/index.js [hunk 3]
```diff
@@ -199,15 +200,16 @@ export default class BlogPost extends React.Component {
   addSiblingsList(showSiblingArticlesList, flyTitle, siblingsListTitle, elementClassName, content) {
```
Comment (varholak-peter) [inline src/index.js]: I know this is outside of your change, but if we could make this stateless, that would be nice.

Comment (varholak-peter) [inline src/index.js]: To clarify, you are getting data from the props inside the function, you can require them as parameters and pass them from props as arguments when calling the function, making the function pure and easier to debug.

**Comment by umbobabo** [comment]
Do you need to update or add more tests?

**Comment by umbobabo** [comment]
@yasgreen93 @garySiladi @varholak-peter You can undestand if your code is covered enough running 'npm run posttest' that produce a tests coverage.

**Pull request merged**